src/OpenSage.Mods.Generals/Gui/GeneralsControlBar.cs

```diff
@@ -0,0 +1,285 @@
+                {
+                    control.Hide();
+                }
+
+                foreach (var control in controlBar._right.Controls)
+                {
+                    control.Hide();
+                }
+            }
+
+            public override void Update(Player player, GeneralsControlBar controlBar)
+            {
+
+            }
+        }
+
+        private sealed class SelectedControlBarState : ControlBarState
+        {
+            public override void OnEnterState(GeneralsControlBar controlBar)
+            {
+                throw new System.NotImplementedException();
+            }
+
+            public override void Update(Player player, GeneralsControlBar controlBar)
+            {
+                throw new System.NotImplementedException();
+            }
+        }
+
+        private sealed class UnderConstructionControlBarState : ControlBarState
+        {
+            public override void OnEnterState(GeneralsControlBar controlBar)
+            {
+                throw new System.NotImplementedException();
+            }
+
+            public override void Update(Player player, GeneralsControlBar controlBar)
+            {
+                throw new System.NotImplementedException();
+            }
+        }
+    }
+
+    public sealed class GeneralsControlBarSource : IControlBarSource
+    {
+        public IControlBar Create(string side, ContentManager contentManager)
+        {
+            // TODO: This is not the best place for this.
+            contentManager.IniDataContext.LoadIniFile(@"Data\INI\ControlBarScheme.ini");
+
+            var scheme = contentManager.IniDataContext.ControlBarSchemes.FindBySide(side);
+
+            // TODO: Support multiple image parts?
+            // Generals always uses exactly one image part.
+            var imagePart = scheme.ImageParts[0];
+
+            var background = new Control
+            {
+                Name = "OpenSAGE:ControlBarBackground",
+                Bounds = new Rectangle(imagePart.Position, imagePart.Size),
+                BackgroundImage = LoadImage(imagePart.ImageName)
+            };
+
+            var backgroundWindow = new Window(scheme.ScreenCreationRes, background, contentManager);
+            var controlBarWindow = contentManager.Load<Window>("Window/ControlBar.wnd", new LoadOptions { CacheAsset = false });
+
+            Control FindControl(string name) => controlBarWindow.Controls.FindControl($"ControlBar.wnd:{name}");
+            Image LoadImage(string path) => contentManager.WndImageLoader.CreateNormalImage(path);
+
+            // TODO: Implement under attack indicator.
+            FindControl("WinUAttack").Hide();
+
+            // TODO: What is this?
+            FindControl("OnTopDraw").Hide();
+
+            var windowOrigin = controlBarWindow.Bounds.Location;
+            var schemeType = scheme.GetType();
+
+            Control ApplyBounds(string name, string coordPrefix)
```

Comment:
I normally wouldn't use reflection for something that can be done safely without it, but the amount of boilerplate this saves is pretty huge. And since this is only called during initialisation, the performance difference is negligible.